src/ScriptBuilder/Writers/OutboxWriter.cs

```diff
@@ -1,21 +1,41 @@
+using System;
 using System.IO;
 using NServiceBus.Persistence.Sql.ScriptBuilder;
 
 class OutboxWriter
 {
-    public static void WriteOutboxScript(string scriptPath, BuildSqlDialect sqlDialect)
+    public static void WriteOutboxScript(bool clean, bool overwrite, string scriptPath, BuildSqlDialect sqlDialect)
     {
         var createPath = Path.Combine(scriptPath, "Outbox_Create.sql");
-        File.Delete(createPath);
+        if (clean)
+        {
+            File.Delete(createPath);
+        }
+
+        CheckExistingFile(overwrite, createPath);
         using (var writer = File.CreateText(createPath))
         {
             OutboxScriptBuilder.BuildCreateScript(writer, sqlDialect);
         }
+
         var dropPath = Path.Combine(scriptPath, "Outbox_Drop.sql");
-        File.Delete(dropPath);
+        if (clean)
+        {
+            File.Delete(dropPath);
+        }
+
+        CheckExistingFile(overwrite, dropPath);
         using (var writer = File.CreateText(dropPath))
         {
             OutboxScriptBuilder.BuildDropScript(writer, sqlDialect);
         }
     }
+
+    static void CheckExistingFile(bool overwrite, string sagaFileName)
```

Comment:
`sagaFileName` should be `outboxFileName`. Maybe this should just be `fileName`?

Reply:
won't be applicable after the refactoring to the base class.